backend/open_webui/models/prompts.py

```
@@ -155,5 +313,127 @@ def delete_prompt_by_command(self, command: str) -> bool:
+                .all()
+            )
+
+            # For the small result set, check shared prompts with has_access
+            accessible_prompts = []
+            for prompt in prompts:
+                # Public prompts (access_control is stored as string 'null') are accessible to everyone
+                if prompt.access_control == "null" or prompt.access_control is None:
+                    accessible_prompts.append(prompt)
+                # User's own prompts are always accessible
+                elif prompt.user_id == user_id:
+                    accessible_prompts.append(prompt)
```

Ruff (0.12.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)